components/scream/scripts/gather_all_data.py

```diff
@@ -86,7 +86,7 @@ def formulate_command(self, machine):
         if machine in ["waterman", "white", "lassen"] and "OMPI_CXX" not in " ".join(env_setup):
             extra_env = "OMPI_CXX={}/bin/nvcc_wrapper ".format(kokkos_loc)
         else:
-            extra_env = "OMP_PROC_BIND=FALSE "
+            extra_env = "OMP_PROC_BIND=FALSE KMP_AFFINITY=balanced "
```

Comment:
Not all machines will use intel (we may test on P9 with just gcc). Perhaps we should set both KMP_AFFINITY _and_ OMP_PROC_BIND=spread. The former overrules the latter with intel, while it's ignored on non-intel builds.